### PR TITLE
ref(ui): Replace old editableText with the new component

### PR DESCRIPTION
**Pull request by priscilawebdev**
Discover

**Before:**
image

image

**After:**

image

Dashboards

Before:
image

After:

image

### CODE REVIEW

File: static/app/views/dashboardsV2/title.tsx [hunk 1]
```diff
@@ -1,87 +1,42 @@
...
+    <Container>
+      {!dashboard ? (
+        t('Dashboards')
+      ) : (
+        <EditableText
           value={dashboard.title}
-          onBlur={this.onBlur}
+          onChange={newTitle => onUpdate({...dashboard, title: newTitle})}
+          errorMessage={t('Please set a title for this widget')}
+          successMessage={t('Widget title saved successfully')}
```
Comment (markstory) [inline static/app/views/dashboardsV2/title.tsx:26]: Isn't this the title for the entire dashboard?

Comment (markstory) [inline static/app/views/dashboardsV2/title.tsx:26]: Isn't this the title for the entire dashboard?

File: static/app/views/dashboardsV2/widget/header.tsx [hunk 2]
```diff
@@ -27,7 +27,12 @@ function Header({title, orgSlug, onChangeTitle}: Props) {
           ]}
         />
         <Layout.Title>
-          <EditableText value={title} onChange={onChangeTitle} />
+          <EditableText
+            value={title}
+            onChange={onChangeTitle}
+            errorMessage={t('Please set a title for this dashboard')}
+            successMessage={t('Dashboard title saved successfully')}
```
Comment (markstory) [inline static/app/views/dashboardsV2/widget/header.tsx:34]: This should be 'dashboard widget title'

**Review by markstory — COMMENTED** [review]
A few notes:

* The pencil icon for editable text doesn't have a pointer cursor.
* The text always has a faint underline. Should that be there? 
Screen Shot 2021-04-16 at 10 42 19 AM
* Because of the padding on the input left alignment with other text is no longer correct 
Screen Shot 2021-04-16 at 10 43 21 AM

File: static/app/views/dashboardsV2/title.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/dashboardsV2/title.tsx:26]: good catch :)

**Comment by priscilawebdev** [comment]
thanks for the feedback @markstory  I have implemented your suggestions 

feedback

File: static/app/views/dashboardsV2/widget/header.tsx [hunk 3]
```diff
@@ -27,7 +27,12 @@ function Header({title, orgSlug, onChangeTitle}: Props) {
           ]}
         />
         <Layout.Title>
-          <EditableText value={title} onChange={onChangeTitle} />
+          <EditableText
+            value={title}
+            onChange={onChangeTitle}
+            errorMessage={t('Please set a title for this dashboard widge')}
```
Comment (markstory) [inline static/app/views/dashboardsV2/widget/header.tsx:33]:
```suggestion
            errorMessage={t('Please set a title for this dashboard widget')}
```

File: static/app/views/dashboardsV2/widget/header.tsx [hunk 4]
```diff
@@ -27,7 +27,12 @@ function Header({title, orgSlug, onChangeTitle}: Props) {
...
+            successMessage={t('Dashboard widge title saved successfully')}
```
Comment (markstory) [inline static/app/views/dashboardsV2/widget/header.tsx:34]:
```suggestion
            successMessage={t('Dashboard widget title saved successfully')}
```

**Comment by dashed** [comment]
Good stuff

**Review by markstory — APPROVED** [review]
Looks good to me.

**Review by dashed — APPROVED** [review]

**Pull request merged**